### ISSUE TITLE
Product missing groovy dependencies

### ISSUE THREAD

**Issue by jhnesk** [gretty-gradle-plugin/gretty#238]
With Gretty 3.0.6 when I try to run the built product it fails with exception:

Error: Could not find or load main class org.akhikhl.gretty.GrettyStarter
Caused by: java.lang.NoClassDefFoundError: groovy/lang/GroovyObject

I have tried this with all versions between 3.0.1 and 3.0.6. Both 3.0.5 and 3.0.6 crashes, but they get different but similar exceptions.

Steps to reproduce:
1. Clone https://github.com/gretty-gradle-plugin/gretty-sample
2. Run this on the sample project do make sure it works as expected.
    $ ./gradlew buildProduct && ./build/output/gretty-sample/run.sh
3. Change gretty version:
    $ sed -i 's/3.0.1/3.0.6/' build.gradle
4. Test with step 2 again and it crashes
    $ ./gradlew buildProduct && ./build/output/gretty-sample/run.sh

I'm downgrading to 3.0.4 now and unfortunately I don't have time to investigate further or solve this and submit a PR at the moment.
Hopefully someone more familiar with Gretty can solve the issue faster. I have not yet tried 4.0

**Comment by jhnesk** [comment]
@boris-petrov Yes, I understand gretty-sample seems more or less unmaintained, I got this error in my own project, but used gretty-sample to make it easy for you to reproduce.
Thank you for looking into the issue.

**Comment by f4lco** [comment]
@boris-petrov That seems very reasonable. I'll have a look at the reproducer and the exact use-cases of the method you mentioned.
@jhnesk thank you for your report!
